lib/jsonapi/utils/string.ex

```diff
@@ -109,12 +112,19 @@ defmodule JSONAPI.Utils.String do
     with words <-
            Regex.split(
              ~r{(?<=[a-zA-Z0-9])[-_](?=[a-zA-Z0-9])},
-             to_string(value)
+             to_string(value),
+             trim: true
            ) do
-      [h | t] = words |> Enum.filter(&(&1 != ""))
-
-      [String.downcase(h) | camelize_list(t)]
-      |> Enum.join()
+      case words do
+        # If there is only one word, leave it as-is
+        [word] ->
+          word
```

Comment:
Annotation: This is the main change.  Like the comment says, if we could not split the value into multiple words (i.e. there were no `-` or `_` delimiters) then return the single word without downcasing it.